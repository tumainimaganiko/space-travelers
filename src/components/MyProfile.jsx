import { useSelector } from 'react-redux';
import Styles from '../styles/Mission.module.css';
import styles from '../styles/MyProfile.module.css';

const MyProfile = () => {
  const { missions } = useSelector((store) => store.missions);
  const bookedMission = missions.filter((mission) => mission.reserved);
  const { rockets } = useSelector((store) => store.rockets);
  const bookedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <>
      <div className={styles.missions}>
        <h2>My Missions</h2>
        <ul>
          {bookedMission.length ? (
            bookedMission.map((mission) => (
              <li key={mission.mission_id} className={Styles.profileRow}>
                {mission.mission_name}
              </li>
            ))
          ) : (
            <p>There are no booked missions</p>
          )}
        </ul>
      </div>
      <div className={styles.rockets}>
        <h2>My Rockets</h2>
        <ul>
          {bookedRockets.length ? (
            bookedRockets.map((rocket) => (
              <li key={rocket.id}>{rocket.name}</li>
            ))
          ) : (
            <p>There are no booked rockets</p>
          )}
        </ul>
      </div>
    </>
  );
};

export default MyProfile;
