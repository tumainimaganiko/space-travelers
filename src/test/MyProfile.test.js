import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import myRockets from './rocketData';
import MyProfile from '../components/MyProfile';
import myMissions from './missionsData';

jest.mock('axios');

const reducer = (
  state = {
    missions: { missions: myMissions },
    rockets: { rockets: myRockets },
  },
) => state;

const store = configureStore({ reducer });

describe('Test MyProfile component', () => {
  it('shouls show reserved Rockets and Missions', () => {
    axios.get.mockResolvedValue({ data: myMissions });
    axios.get.mockResolvedValue({ data: myRockets });
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <MyProfile />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
