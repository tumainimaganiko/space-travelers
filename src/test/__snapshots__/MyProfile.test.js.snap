// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyProfile component shouls show reserved Rockets and Missions 1`] = `
Array [
  <div
    className="missions"
  >
    <h2>
      My Missions
    </h2>
    <ul>
      <p>
        There are no booked missions
      </p>
    </ul>
  </div>,
  <div
    className="rockets"
  >
    <h2>
      My Rockets
    </h2>
    <ul>
      <p>
        There are no booked rockets
      </p>
    </ul>
  </div>,
]
`;
